src/Tests/dotnet-new.Tests/DotnetClassTemplateTests.cs

```diff
@@ -0,0 +1,104 @@
+        private readonly ILogger _logger;
+
+        public DotnetClassTemplateTests(SharedHomeDirectory fixture, ITestOutputHelper log) : base(log)
+        {
+            _fixture = fixture;
+            _log = log;
+            _logger = new XunitLoggerProvider(log)
+                .CreateLogger("TestRun");
+        }
+
+        [Theory]
+        [InlineData("class")]
+        [InlineData("class", "9.0", "netstandard2.0")]
+        [InlineData("interface")]
+        [InlineData("interface", "9.0", "netstandard2.0")]
+        [InlineData("record")]
+        [InlineData("record", "9.0")]
+        [InlineData("record", "8.0", "netstandard2.0")]
+        [InlineData("struct")]
+        [InlineData("struct", "9.0", "netstandard2.0")]
+        [InlineData("enum")]
+        [InlineData("enum", "9.0", "netstandard2.0")]
+        public async void DotnetCSharpClassTemplatesTest(
+            string templateShortName,
+            string langVersion = "preview",
+            string targetFramework = "net7.0")
+        {
+            // prevents logging welcome message from sdk installation
+            Dictionary<string, string> environmentUnderTest = new() { ["DOTNET_NOLOGO"] = false.ToString() };
+
+            string expectedProjectName = $"{templateShortName}.langVersion={langVersion}.targetFramework={targetFramework}";
+            string workingDir = CreateTemporaryFolder($"{nameof(DotnetCSharpClassTemplatesTest)}.{expectedProjectName}");
+            string projectName = CreateTestProject(workingDir, langVersion, targetFramework);
+
+            TemplateVerifierOptions options = new TemplateVerifierOptions(templateName: templateShortName)
+            {
+                SnapshotsDirectory = "Approvals",
+                VerifyCommandOutput = true,
+                TemplateSpecificArgs = new[] { "--name", "Class1" },
+                VerificationExcludePatterns = new[]
+                {
+                    "*/stderr.txt",
+                    "*\\stderr.txt",
+                    $"*{projectName}.*",
+                    "*project.*.*"
+                },
+                SettingsDirectory = _fixture.HomeDirectory,
+                DotnetExecutablePath = TestContext.Current.ToolsetUnderTest.DotNetHostPath,
+                DoNotAppendTemplateArgsToScenarioName = true,
+                DoNotPrependTemplateNameToScenarioName = true,
+                ScenarioName = expectedProjectName,
+                OutputDirectory = workingDir,
+                EnsureEmptyOutputDirectory = false
+            }
+            .WithCustomEnvironment(environmentUnderTest)
+            .WithCustomScrubbers(
+                ScrubbersDefinition.Empty
+                    .AddScrubber(sb => sb.Replace($"_{projectName}", $"{expectedProjectName}")));
+
+            VerificationEngine engine = new VerificationEngine(_logger);
+            await engine.Execute(options)
+                .ConfigureAwait(false);
+        }
+
+        private string CreateTestProject(
```

Comment:
Is this run just to get the project name - that sounds overkill. Is there any alternative?

Reply:
It runs for the generation of a project file (csproj).
Without it and having it restored we can't use class templates